stock_account_operating_unit/tests/test_stock_account_operating_unit.py

```diff
@@ -0,0 +1,365 @@
+        """Create a user."""
+        group_ids = [group.id for group in groups]
+        user = self.res_users_model.create(
+            {
+                "name": "Test Stock Account User",
+                "login": login,
+                "password": "demo",
+                "email": "example@yourcompany.com",
+                "company_id": company.id,
+                "company_ids": [(4, company.id)],
+                "operating_unit_ids": [(4, ou.id) for ou in operating_units],
+                "groups_id": [(6, 0, group_ids)],
+            }
+        )
+        return user
+
+    def _create_account(self, acc_type, name, code, company):
+        """Create an account."""
+        account = self.account_model.create(
+            {
+                "name": name,
+                "code": code,
+                "user_type_id": acc_type.id,
+                "company_id": company.id,
+            }
+        )
+        return account
+
+    def _create_product(self):
+        """Create a Product with inventory valuation set to auto."""
+        product_cteg = self.product_cteg_model.create(
+            {
+                "name": "test_product_ctg",
+                "property_valuation": "real_time",
+                "property_stock_valuation_account_id": self.account_inventory.id,
+                "property_stock_account_input_categ_id": self.account_grni.id,
+                "property_stock_account_output_categ_id": self.account_cogs_id,
+            }
+        )
+        product = self.product_model.create(
+            {
+                "name": "test_product",
+                "categ_id": product_cteg.id,
+                "type": "product",
+                "list_price": 1.0,
+                "standard_price": 1.0,
+            }
+        )
+        return product
+
+    def _create_picking(self, user, ou_id, picking_type, src_loc_id, dest_loc_id):
+        """Create a Picking."""
+        picking = self.picking_model.with_user(user.id).create(
+            {
+                "picking_type_id": picking_type.id,
+                "location_id": src_loc_id.id,
+                "location_dest_id": dest_loc_id.id,
+                "operating_unit_id": ou_id.id,
+            }
+        )
+        self.move_model.with_user(user.id).create(
+            {
+                "name": "a move",
+                "product_id": self.product.id,
+                "product_uom_qty": 1.0,
+                "product_uom": 1,
+                "picking_id": picking.id,
+                "location_id": src_loc_id.id,
+                "location_dest_id": dest_loc_id.id,
+            }
+        )
+        return picking
+
+    def _confirm_receive(self, user_id, picking):
+        """
+        Checks the stock availability, validates and process the stock picking.
+        """
+        picking.action_confirm()
+        picking.action_assign()
+        res = picking.with_user(user_id).button_validate()
+        validate_id = res["res_id"]
```

Comment:
Test is failing here please fix:
stock_account_operating_unit/tests/test_stock_account_operating_unit.py", line 177, in _confirm_receive
    validate_id = res["res_id"]
KeyError: 'res_id'